compiler-rt/lib/sanitizer_common/sanitizer_posix_libcdep.cpp

```diff
@@ -416,8 +416,19 @@ bool MmapFixedSuperNoReserve(uptr fixed_addr, uptr size, const char *name) {
 }
 
 uptr ReservedAddressRange::Init(uptr size, const char *name, uptr fixed_addr) {
+  // AIX can not mmap on a mmaped memory, so init it as read/write so we won't
+  // mmap on this memory again.
+#  if SANITIZER_AIX
+  if (fixed_addr) {
+    MmapFixed(fixed_addr, size, MAP_PRIVATE | MAP_FIXED | MAP_ANON, name);
+    base_ = (void *)fixed_addr;
+  } else
+    base_ = (void *)internal_mmap(nullptr, size, PROT_READ | PROT_WRITE,
+                                  MAP_PRIVATE | MAP_ANON, -1, 0);
```

Comment:
Nit: See https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements example for:
```
// Use braces for the `else if` and `else` block to keep it uniform with the
// `if` block.
```
```suggestion
  } else {
    base_ = (void *)internal_mmap(nullptr, size, PROT_READ | PROT_WRITE,
                                  MAP_PRIVATE | MAP_ANON, -1, 0);
  }
```